connmgr/seed.go

```diff
@@ -29,8 +30,17 @@ type OnSeed func(addrs []*wire.NetAddress)
 type LookupFunc func(string) ([]net.IP, error)
 
 // SeedFromDNS uses DNS seeding to populate the address manager with peers.
-func SeedFromDNS(chainParams *chaincfg.Params, lookupFn LookupFunc, seedFn OnSeed) {
+func SeedFromDNS(chainParams *chaincfg.Params, reqServices wire.ServiceFlag,
+	lookupFn LookupFunc, seedFn OnSeed) {
+
 	for _, seeder := range chainParams.DNSSeeds {
+		var host string
+		if !seeder.HasFiltering || reqServices == wire.SFNodeNetwork {
+			host = seeder.Host
+		} else {
+			host = fmt.Sprintf("x%x.%s", reqServices, seeder.Host)
+		}
+
 		go func(seeder string) {
```

Comment:
Should this make the argument match the name passed to it or choose a different name altogether?  It seems to me like it might be rather easier to confuse this with the new `DNSSeed` struct in the code below due to it having the same name as the variable in the outer scope introduced by the for loop over `chainParams.DNSSeeds`.